packages/core/src/embeddings/types.ts

```diff
@@ -21,4 +27,32 @@ export abstract class BaseEmbedding {
 
   abstract getTextEmbedding(text: string): Promise<number[]>;
   abstract getQueryEmbedding(query: string): Promise<number[]>;
+
+  async getTextEmbeddingBatch(texts: string[]): Promise<Array<Embedding>> {
```

Comment:
1. How about having a default implementation in `BaseEmbedding` for the batching that just falls back and uses the non-batching method?
Then the consumers of `getTextEmbedding` can be changed to use batching generally - I think it's just `VectorStoreIndex`.
2. How about overloading `getTextEmbedding` instead?
`getTextEmbedding(texts: string[]): Promise<number[][]>;`

Reply:
@marcusschiesser nice catches! can you expand a bit on the second one?

> How about overloading getTextEmbedding instead?
getTextEmbedding(texts: string[]): Promise<number[][]>;

Reply:
I think for now it would case more breaking changes, and personally I think that's a bit confuse